internal/gcs/bridge.go

```diff
@@ -302,6 +305,11 @@ func (brdg *bridge) recvLoop() error {
 			if err == io.EOF || isLocalDisconnectError(err) { //nolint:errorlint
 				return nil
 			}
+			if errors.Is(err, errMessageSize) {
+				// should be safe to only warn and not crash the bridge if the message size is wrong
+				brdg.log.WithError(err).Warn("bridge received failed")
+				continue
```

Comment:
shouldn't we also `call.complete(err)` as well? otherwise, we'll never clear `brgs.rpcs[id]`, essentially leaking it? also, would the RPC call just hang in this case too? Maybe it's worth adding this functionality in a separate PR and just handle the message size trimming for now.